File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

bool compare(int& a, int& b) {
    return a >b;
}

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    map<int, int> m;
    
    for(int& item : tangerine)
    {
        if(m.find(item) != m.end()) m[item] +=1;
        else m.insert({item, 1});
    }
    
    vector<int> vec;
    
    for (auto it = m.begin(); it != m.end(); it++) 
        vec.push_back(it->second);
    
    sort(vec.begin(), vec.end(), compare);
    
    for(auto& item : vec)
    {
        answer +=1;
        if(item >= k) break;
        else k -= item;
    }
    
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    
    let amount = {};
    
    for(const item of tangerine) {
        if(amount[item])amount[item] +=1;
        else amount[item] = 1;
    }
    
    let result = [];
    
    for(const key in amount) {
        result.push([key, amount[key]]);
    }
    
    result.sort((a,b) => {
        return b[1] - a[1];
    })
    
    for(const item of result) {   
        if(item[1] >= k) {
            answer +=1;
            break;
        } else{
            k -= item[1];
            answer +=1;
        }
    }
    
    
    
    return answer;
}